testsuite/tests/typing-layouts-arrays/array_element_size_in_bytes.ml

```diff
@@ -8,20 +8,53 @@
  }
 *)
 
-(* CR layouts v4: The below definition is just to give this test slightly
-   different behavior on native code and bytecode, because some arrays of
-   unboxed things are represented as custom blocks on only native code, and
-   therefore the size calculations differ slightly. Delete this when we change
-   the representation to not use custom blocks. *)
-let custom_block_padding =
-  match Sys.backend_type with
-  | Native -> 1
-  | Bytecode -> 0
-  | Other _ -> failwith "Don't know what to do"
+(* Unboxed arrays now use normal blocks instead of custom blocks *)
 
 (* We only compile for 64 bits. *)
 let bytes_per_word = 8
 
+(* Tag definitions from Cmm_helpers.Unboxed_array_tags *)
+let unboxed_product_array_tag = 0
+let unboxed_int64_array_tag = 1
+let unboxed_int32_array_even_tag = 2
+let unboxed_int32_array_odd_tag = 3
+let unboxed_float32_array_even_tag = 4
+let unboxed_float32_array_odd_tag = 5
+let unboxed_vec128_array_tag = 6
+let unboxed_vec256_array_tag = 7
+let unboxed_vec512_array_tag = 8
+let unboxed_nativeint_array_tag = 9
```

Comment:
I believe there are two ways of creating arrays (via the runtime and at compile time??) Do these tests only exercise the runtime, or do they or something else cover the other side?